core/sequencer/server.go

```diff
@@ -236,12 +236,17 @@ func (s *Server) DefineRevisions(ctx context.Context,
 		HighestApplied: revs.HighestApplied,
 		HighestDefined: revs.HighestDefined,
 	}
-	// Don't create new revisions if there are ones waiting to be applied, or the
-	// highest defined revision is lagging behind for some reason.
-	if resp.HighestDefined != resp.HighestApplied {
+	// Do nothing if the highest defined revision is lagging behind for some
+	// reason. It will catch up later.
+	if resp.HighestDefined < resp.HighestApplied {
 		return resp, nil
 	}
-	// TODO(#1056): Allow going ahead to some extent.
+	// Allow at most one pending revision, which enables applying a revision and
+	// defining the next one simultaneously.
+	if resp.HighestDefined > resp.HighestApplied+1 {
```

Comment:
Could you pull this out into a constant or a flag? I can imagine this being a value we'd want to tweek for performance